composition/html_content_parser.go

```diff
@@ -377,7 +378,7 @@ forloop:
 	s := headBuff.String()
 
 	if len(s) > 0 {
-		*fragment = *NewStringFragment(s)
+		fragment.SetContent(s)
```

Comment:
it would be better to refactor the ugly method signature above, in the way, that no dangerous pointer stuff is needed!